overlord/fdestate/fdemgr.go

```diff
@@ -300,6 +310,79 @@ func (m *FDEManager) GetParameters(role string, containerRole string) (hasParame
 	return s.getParameters(role, containerRole)
 }
 
+const recoveryKeyExpireAfter = 5 * time.Minute
+
+func (m *FDEManager) generateRecoveryKey() (rkey keys.RecoveryKey, keyID string, err error) {
+	if m.recoveryKeyCache == nil {
+		return keys.RecoveryKey{}, "", errors.New("internal error: recoveryKeyCache is nil")
+	}
+
+	rkey, err = keysNewRecoveryKey()
+	if err != nil {
+		return keys.RecoveryKey{}, "", err
+	}
+
+	rkeyInfo := backend.CachedRecoverKey{
+		Key:        rkey,
+		Expiration: timeNow().Add(recoveryKeyExpireAfter),
+	}
+
+	var lastRecoveryKeyID int
+	err = m.state.Get("last-recovery-key-id", &lastRecoveryKeyID)
```

Comment:
Could we use a hash of the key rather than keeping a counter in state?

Reply:
it's indeed a bit strange to use a counter in state for the ids of something that exist only in memory, you could use either what Valentin suggest, though probably you would want to use just a prefix and be slightly careful about the unlikely clashes,  or use an increasing number in state Cache that you initialise with a something like a combo of time + some random bits

Reply:
Great idea, implemented hash prefix key-ids, also added some simple collision detection and reasonable retries.